agent/consul/state/catalog.go

```diff
@@ -3278,15 +3278,23 @@ func cleanupMeshTopology(tx WriteTxn, idx uint64, service *structs.ServiceNode)
 		if !ok {
 			return fmt.Errorf("unexpected topology type %T", rawCopy)
 		}
-		delete(copy.Refs, uid)
+		if _, ok := copy.Refs[uid]; !ok {
+			continue
```

Comment:
I guess this could be done on the `entry` so that we skip the deep copy in this case? Probably a minor optimization.